scripts/lib/compare-statements.test.js

```diff
@@ -0,0 +1,112 @@
+/* This file is a part of @mdn/browser-compat-data
+ * See LICENSE file for more information. */
+
+import assert from 'node:assert/strict';
+
+import compareStatements from './compare-statements.js';
+
+const tests = [
+  {
+    input: {
+      __compat: {
+        support: {
+          chrome: [
+            { version_added: '20', prefix: 'webkit' },
+            {
+              version_added: '10',
+              version_removed: '18',
+              partial_implementation: true,
+              notes: 'No fries with the burger',
+            },
+            {
+              version_added: '12',
+              flags: [
+                {
+                  type: 'preference',
+                  name: '#add_fries',
+                  value_to_set: 'Extra Crispy',
+                },
+              ],
+            },
+            { version_added: '20' },
+          ],
+        },
+      },
+    },
+    output: {
+      __compat: {
+        support: {
+          chrome: [
+            { version_added: '20' },
+            {
+              version_added: '10',
+              version_removed: '18',
+              partial_implementation: true,
+              notes: 'No fries with the burger',
+            },
+            { version_added: '20', prefix: 'webkit' },
```

Comment:
This is a bit surprising to me that this active support item comes after a chronologicaly older inactive (version_removed) item. 😕

Reply:
The idea was that prefixes would be moved further down in the list, but I see what you mean -- let me update this!